STYLE.md

```diff
@@ -30,7 +30,8 @@
   fiasco](https://isocpp.org/wiki/faq/ctors#static-init-order-on-first-use) for
   how to best handle this.
 * API-level comments should follow normal Doxygen conventions. Use `@param` to describe
-  parameters, `@return <return-type>` for return values.
+  parameters, `@return <return-type>` for return values. Internal comments for
```

Comment:
I think I'd still prefer that we just relax the requirement for "going full doxygen" for private docs rather than mandating that private member/method have comments that Doxygen must ignore.  Even in your example "foo does bar" is a perfectly reasonable comment for Doxygen to pick up as an internal method, extensive return type documentation isn't mandatory.  In other words: Suggest but not require that private members/methods have a doxygen style comment, and make no hard mandates about all doxygen metadata being populated.

But, I like documenting things pretty extensively as part of my mental code writing process so that's my bias...